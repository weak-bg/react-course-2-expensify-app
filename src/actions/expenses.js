import uuid from 'uuid';

//add_expense
export const addExpense = (
    {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

//remove_expense
export const removeExpense = ({id} ={}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

//edit_expense
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});