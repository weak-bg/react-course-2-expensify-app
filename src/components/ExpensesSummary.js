import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
    {
        props.expenses.length <= 1 ? (
            <p>
                Viewing {props.expenses.length} expense totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}
            </p>
        ): (
            <p>
                Viewing {props.expenses.length} expenses totalling {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}
            </p> 
        )
}
    </div>
);

const mapStatetoProps = (state) => {
    return{
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStatetoProps)(ExpensesSummary);