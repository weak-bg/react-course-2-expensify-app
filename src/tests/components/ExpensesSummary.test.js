import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

test('should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with many expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});