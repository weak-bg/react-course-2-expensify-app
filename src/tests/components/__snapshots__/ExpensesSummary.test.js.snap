// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense totalling 
    $1,095.00
  </p>
</div>
`;

exports[`should render ExpensesSummary with many expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,141.95
  </p>
</div>
`;
